src/covid-vaccine-slot-booking.py

```diff
@@ -122,13 +122,10 @@ def main():
 
                 tryOTP = input('Try for a new Token? (y/n Default y): ')
                 if tryOTP.lower() == 'y' or not tryOTP:
-                    if mobile:
-                        token = generate_token_OTP(mobile, base_request_header)
-                        token_valid = True
-                    else:
+                    if mobile is None:
                         mobile = input("Enter the registered mobile number: ")
```

Comment:
That won't be needed since the code asks for mobile number only if it is empty. If not empty, it means that the mobile number is already present and thereby can directly invoke the OTP API. The change I made is to refactor the code ... ever so slightly ... to move asking for mobile number into a condition. After that the code is common. Hope this clarifies.